sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/PartitionReceiverOptions.cs

```diff
@@ -0,0 +1,182 @@
+        private int _prefetchCount = 300;
+
+        /// <summary>
+        ///   The options used for configuring the connection to the Event Hubs service.
+        /// </summary>
+        ///
+        public EventHubConnectionOptions ConnectionOptions
+        {
+            get => _connectionOptions;
+            set
+            {
+                Argument.AssertNotNull(value, nameof(ConnectionOptions));
+                _connectionOptions = value;
+            }
+        }
+
+        /// <summary>
+        ///   The set of options to use for determining whether a failed operation should be retried and,
+        ///   if so, the amount of time to wait between retry attempts.  These options also control the
+        ///   amount of time allowed for reading events and other interactions with the Event Hubs service.
+        /// </summary>
+        ///
+        public EventHubsRetryOptions RetryOptions
+        {
+            get => _retryOptions;
+            set
+            {
+                Argument.AssertNotNull(value, nameof(RetryOptions));
+                _retryOptions = value;
+            }
+        }
+
+        /// <summary>
+        ///   The default amount of time to wait for the requested amount of messages when reading; if this
+        ///   period elapses before the requested amount of messages were available or read, then the set of
+        ///   messages that were read will be returned.
+        /// </summary>
+        ///
+        public TimeSpan? DefaultMaximumReceiveWaitTime
+        {
+            get => _defaultMaximumReceiveWaitTime;
+
+            set
+            {
+                if (value.HasValue)
+                {
+                    Argument.AssertNotNegative(value.Value, nameof(DefaultMaximumReceiveWaitTime));
+                }
+
+                _defaultMaximumReceiveWaitTime = value;
+            }
+        }
+
+        /// <summary>
+        ///   When populated, the owner level indicates that a reading is intended to be performed exclusively for events in the
+        ///   requested partition and for the associated consumer group.  To do so, reading will attempt to assert ownership
+        ///   over the partition; in the case where more than one exclusive reader attempts to assert ownership for the same
+        ///   partition/consumer group pair, the one having a larger <see cref="OwnerLevel"/> value will "win."
+        ///
+        ///   When an exclusive reader is used, other readers which are non-exclusive or which have a lower owner level will either
+        ///   not be allowed to be created, if they already exist, will encounter an exception during the next attempted operation.
+        /// </summary>
+        ///
+        /// <value>The relative priority to associate with an exclusive reader; for a non-exclusive reader, this value should be <c>null</c>.</value>
+        ///
+        /// <seealso cref="EventHubsException.FailureReason.ConsumerDisconnected"/>
```

Comment:
We probably should expand on this a bit to something like:

```
/// <remarks>
///  A <see cref="EventHubsException">will occur if a <see cref="PartitionReceiver" /> unable to read events from the requested Event Hub partition for the 
///  given consumer group.  In this case the <see cref="EventHubsException.FailureReason"/> will be set to <see cref="EventHubsException.FailureReason.ConsumerDisconnected"/>.
/// </remarks>
```
And then include the `<seealso` for the exception and the one that you have for the reason.